### ISSUE TITLE
Remove dependence on Python

### ISSUE THREAD

**Issue by malmaud** [malmaud/TensorFlow.jl#20]
This relies on the C API supporting adding gradient nodes, which Google has said will happen sometime.

**Comment by spencerlyon2** [comment]
Any timeline on this?

**Comment by malmaud** [comment]
No. There is a just TODO in the readme for the C API that says to add support support for gradient calculations.

**Comment by MikeInnes** [comment]
In the mean time, would you be amenable to splitting out, say, TensorFlowCore.jl to provide the core API and have TensorFlow.jl wrap that + provide extras like `gradients`? It'd be pretty useful for applications that don't need `gradients` not to have the hard python dependency.

Alternatively, perhaps we could do something like lazy-loading the python process when `gradients` is first called?

**Comment by malmaud** [comment]
I'd be up for lazy-loading the python process. In practice I would think most uses of TensorFlow would use `gradients`, but if nothing else, you'd get a faster start-up time.

**Comment by oxinabox** [comment]
I think this might be unblocked now:
https://github.com/tensorflow/tensorflow/issues/6268

**Comment by malmaud** [comment]
It is is closer, but still far away - many gradients are still missing from the C API.
